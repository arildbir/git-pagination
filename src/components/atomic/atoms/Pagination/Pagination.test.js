import React from "react";
import View from "./Pagination";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const minProps = {
  loading: false,
  numberOfPages: 10,
  currentPage: 2
};

describe("Pagination", () => {
  it("renders", () => {
    const wrapper = shallow(<View {...minProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h3").text()).toEqual("change page");
  });

  it("one button has selected style", () => {
    const wrapper = shallow(<View {...minProps} />);
    expect(
      wrapper
        .find("ul")
        .childAt(1)
        .find("button")
        .hasClass("isSelected")
    ).toEqual(true);
  });
});
