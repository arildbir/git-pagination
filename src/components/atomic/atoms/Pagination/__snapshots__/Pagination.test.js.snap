// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders 1`] = `ShallowWrapper {}`;
